sdk/go/common/resource/plugin/provider.go

```diff
@@ -76,35 +76,55 @@ type ProviderHandshakeResponse struct {
 	SupportsAutonamingConfiguration bool
 }
 
+// ParameterizeParameters can either be of concrete type ParameterizeArgs or ParameterizeValue, for when parameterizing
+// a provider from either the command line or from within a Pulumi program, respectively.
 type ParameterizeParameters interface {
 	isParameterizeParameters()
 }
 
 type (
+	// ParameterizeArgs is used when parameterizing a provider from command line arguments.
 	ParameterizeArgs struct {
+		// The arguments passed on the command line following the provider name.
 		Args []string
 	}
 
+	// ParameterizeValue is used when parameterizing a provider from within a Pulumi program.
 	ParameterizeValue struct {
-		Name    string
+		// The name of the parameterization.
+		Name string
+		// The version of the parameterization.
 		Version semver.Version
-		Value   []byte
+		// The provider metadata embedded in the parameterization.
```

Comment:
We simply call this the "parameterization value" everywhere else. I think metadata probably just adds a new word here.

Reply:
This is what all providers call the internal data they use for resolving resource operations. So for provider authors, this should make parameterization a lot more understandable.

Reply:
I've tweaked this to move away from the metadata language.